### PR TITLE
Remove useTheme export

### PR DESCRIPTION
**Pull request by levithomason**
This PR removes the `useTheme` export as it is no longer needed. If you need access to design tokens, import them directly:

```jsx
import { tokens } from '@fluentui/react-components'
```

### CODE REVIEW

File: packages/react-components/react-provider/src/components/FluentProvider/FluentProvider.types.ts [hunk 1]
```diff
@@ -23,7 +24,7 @@ export type FluentProviderProps = Omit<ComponentProps<FluentProviderSlots>, 'dir
 export type FluentProviderState = ComponentState<FluentProviderSlots> &
   Pick<FluentProviderProps, 'targetDocument'> &
   Required<Pick<FluentProviderProps, 'dir'>> & {
-    theme: Theme | Partial<Theme> | undefined;
+    theme: ThemeContextValue;
```
Comment (levithomason) [inline packages/react-components/react-provider/src/components/FluentProvider/FluentProvider.types.ts:27]: There were 5 or 6 instances of this specific typing, so I extracted it.

**Comment by fabricteam** [comment]
## 📊 Bundle size report

<sub>🤖 This report was generated against <a href='https://github.com/microsoft/fluentui/commit/287cf6769bda0ca0ec7cd29b01a4a4481a488d92'>287cf6769bda0ca0ec7cd29b01a4a4481a488d92</a></sub>
<div id="bundleSizeComment9423"/>

**Review by spmonahan — APPROVED** [review]

**Review by miroslavstastny — CHANGES_REQUESTED** [review]
We should go with #23325 instead.

**Comment by levithomason** [comment]
I think this PR is different than https://github.com/microsoft/fluentui/pull/23325. This PR removes the export entirely since it is no longer required due to global context and the ability to import tokens.

The other PR leaves this export but just renames it to "unstable". I think we should merge this one first, then https://github.com/microsoft/fluentui/pull/23325.

File: packages/react-components/react-provider/src/components/FluentProvider/FluentProvider.types.ts (hunk 1 above)
Comment (micahgodbolt) [reply packages/react-components/react-provider/src/components/FluentProvider/FluentProvider.types.ts:27]: Is this related to the useTheme change?

File: packages/react-components/react-shared-contexts/src/index.ts [hunk 2]
```diff
@@ -1,4 +1,5 @@
-export { ThemeContext, useTheme } from './ThemeContext';
+export { ThemeContext } from './ThemeContext';
+export type { ThemeContextValue } from './ThemeContext';
```
Comment (micahgodbolt) [inline packages/react-components/react-shared-contexts/src/index.ts:2]: in https://github.com/microsoft/fluentui/pull/23325 we are moving react-shared-contexts to unstable. Does this have any effect on exporting this type from the unstable package?

**Comment by fabricteam** [comment]
## [Perf Analysis (`@fluentui/react-components`)](https://github.com/microsoft/fluentui/wiki/Perf-Testing)
<p>No significant results to display.</p><details><summary>All results</summary><p>
  <table>
  <tr>
    <th>Scenario</th>
    <th>Render type</th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">Master Ticks</a>
    </th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">PR Ticks</a>
    </th>
    <th>Iterations</th>
    <th>Status</th>
  </tr><tr>
            <td>Avatar</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23335/perf-test-react-components/Avatar-mount_base.html">1225</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23335/perf-test-react-components/Avatar-mount.html">1219</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Button</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23335/perf-test-react-components/Button-mount_base.html">767</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23335/perf-test-react-components/Button-mount.html">737</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProvider</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23335/perf-test-react-components/FluentProvider-mount_base.html">2231</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23335/perf-test-react-components/FluentProvider-mount.html">2219</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23335/perf-test-react-components/FluentProviderWithTheme-mount_base.html">388</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23335/perf-test-react-components/FluentProviderWithTheme-mount.html">393</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23335/perf-test-react-components/FluentProviderWithTheme-virtual-rerender_base.html">355</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23335/perf-test-react-components/FluentProviderWithTheme-virtual-rerender.html">359</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender-with-unmount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23335/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount_base.html">422</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23335/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount.html">405</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>MakeStyles</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23335/perf-test-react-components/MakeStyles-mount_base.html">2024</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23335/perf-test-react-components/MakeStyles-mount.html">1985</a></td>
            <td>50000</td>
            <td></td>
           </tr></table></p></details>

<div id="perfComment9425"/>

**Pull request merged**